### ISSUE TITLE
Switching scenes w/ same storage can corrupt it

### ISSUE THREAD

**Issue by JekSun97** [TokisanGames/Terrain3D#419]
### Terrain3D version

v0.9.3-dev

### System information

Godot v4.2.2.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated Radeon RX 560 Series (Advanced Micro Devices, Inc.; 31.0.14001.45012) - Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz (4 Threads)

### Is the issue reproducable in the demo?

Yes

### Issue description

Suddenly, after another test, the entire terrain system broke down, it stopped being displayed, and there was no collision, restarting the project did not help.

Only the background terrain is displayed

Pre-demonstration of the problem:
https://youtu.be/1Bv2oBoEWlU

Project:
[project.zip](https://github.com/user-attachments/files/16199388/project.zip)

### Logs

empty

**Comment by TokisanGames** [comment]
There are at least two bugs related to when you have multiple scenes open, pointing to the same Terrain3DStorage resource file, possibly with a third scene w/o terrain, all of which you have in your video.

One bug seems to empty region_offsets, which is what happened to your file. You can see it in your inspector.
![image](https://github.com/user-attachments/assets/1287097f-81dd-4880-a631-e11235918dd8)

Another that I hit when looking at your file is related to two files with the same UUID, and Terrain3DStorage, but I can't tell if it's a bug in Godot, godot-cpp, or Terrain3DStorage.

For now, be cautious opening the same file in two scenes and switching between them, especially if going between a non-terrain scene.

You can fix your scene by downloading the demo terrain3D_storage.res file, or by repopulating your region offsets:

In DemoScene.gd add to _ready()
```
	$Terrain3D.storage.set_region_offsets( [Vector2i(0, -2), Vector2i(0, 0), Vector2i(0, -1)] )
```

Save and reload the scene. It should load fixed. Remove the line, and save everything.

**Comment by TokisanGames** [comment]
Two other methods will corrupt storage:
*  Opening Demo, CodeGeneratedDemo, NavigationDemo, then closing them in reverse order will empty region_offsets and mess up storage. (probably what happened in Op)
* Double clicking the reset button next to Material will corrupt storage, with the region_offsets getting emptied. One click at a time is fine, but not double.

Don't run the scene or save after this happens or it will write the changes to disk.

#374 is coming with major changes to storage, so I'm hoping that will address these issues.

**Comment by TokisanGames** [comment]
This has been addressed in #374 and #476. I have not been able to reproduce corrupting the display by switching scenes.
